iOS/Flows/Archives/ArchiveCoordinator.swift

```diff
@@ -115,19 +117,27 @@ extension ArchiveCoordinator: ArchiveViewControllerDelegate {
     }
 
     func startConversationFlow(for conversation: Conversation?, startingMessageId: MessageId?) {
-        self.removeChild()
-
-        let coordinator = ConversationCoordinator(router: self.router,
-                                                  deepLink: self.deepLink,
-                                                  conversation: conversation,
-                                                  startingMessageId: startingMessageId)
-        self.addChildAndStart(coordinator, finishedHandler: { [unowned self] (_) in
-            self.router.dismiss(source: self.archiveVC, animated: true)
-        })
-        self.router.present(coordinator,
-                            source: self.archiveVC,
-                            cancelHandler: {
-        })
+        Task {
+            self.removeChild()
+            guard let conversation = conversation else { return }
+
+            let membersController = ChatClient.shared.memberListController(query: .init(cid: conversation.cid))
+            try? await membersController.synchronize()
+
+            let members = Array(membersController.members)
+
+            let coordinator = ConversationListCoordinator(router: self.router,
+                                                          deepLink: self.deepLink,
+                                                          conversationMembers: members,
+                                                          startingConversationID: conversation.cid)
+            self.addChildAndStart(coordinator, finishedHandler: { [unowned self] (_) in
+                self.router.dismiss(source: self.archiveVC, animated: true)
+            })
+            self.router.present(coordinator,
+                                source: self.archiveVC,
+                                cancelHandler: {
+            })
+        }
```

Comment:
When we start the conversation flow, we get the list of members first. The members controller ensures that the list is complete.

Reply:
Add to taskPool?